MSAL/src/native_auth/controllers/MSALNativeAuthBaseController.swift

```diff
@@ -163,4 +163,11 @@ class MSALNativeAuthBaseController {
             }
         }
     }
+
+    func updateContext(_ previousContext: MSIDRequestContext, with correlationId: UUID?) -> MSALNativeAuthRequestContext {
+        return MSALNativeAuthRequestContext(
+            correlationId: correlationId ?? previousContext.correlationId(),
```

Comment:
Just from the signature of this method I would expect if I passed a nil correlationId that the new context would use it (I know that's not possible). I would prefer if the nil coalescing was done at the call site, or (probably better) if this method was named differently.

Reply:
I'm wondering if it makes more sense to add a new setCorrelationId method in the `MSALNativeAuthRequestContext` class

Reply:
I don't think so because the request's correlation ID was still what it was so updating it doesn't feel right. If we want to store the new correlationId in the `MSALNativeAuthRequestContext` then maybe we could add a new attribute for `responseCorrelationId`. But if all of this is just because we want to use the new correlationId in the error message and logs then maybe we should find a way to pass the values there directly.

Reply:
I've updated it to `updateCorrelationId(newCorrelationId, fromContext: context)`. Let me know if you think a different name would suit it better.

Reply:
@spetrescu84 - Yesterday we talked about a temporary and a final solution for this: 

As a temporary fix, we are going to add to `MSALNativeAuthRequestContext` a variable called `_serverCorrelationId` and a method `setServerCorrelationId()`. This method checks if the initial correlationId and the serverCorrelationId are the same. If they aren't, we log it as a warning, including both correlationIds in the log. And from that moment, every time we call `context.correlationId()`, we return the serverCorrelationId.

But what we are doing is conceptually wrong: we are setting the serverCorrelationId in an object that should be constant, and is actually called `...RequestContext`, not `...serverContext`. So, in the future we will introduce a `FlowContext` object, that will be a wrapper of the `MSALNativeAuthRequestContext` + a server context that we will add. We will discuss the details in future 👍